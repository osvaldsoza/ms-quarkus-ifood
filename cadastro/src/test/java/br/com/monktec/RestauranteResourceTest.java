package br.com.monktec;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testBuscarRestaurantes(){
        String resultado = given()
                .when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
       Approvals.verify(resultado);
    }
}
