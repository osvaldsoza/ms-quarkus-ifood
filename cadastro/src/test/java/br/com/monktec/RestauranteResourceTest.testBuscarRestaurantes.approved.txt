[{"id":123,"nome":"Novo restaurante","proprietario":"ID do keyclock"},{"id":456,"nome":"Novo restaurante 2","proprietario":"ID do keyclock 2"}]